### PR TITLE
TestShell 기본 기능 구현

### PR DESCRIPTION
**Pull request by Kanaish**
TestShell 기본 기능 구현에 대한 PR 입니다.
아래와 같이 5개 tc가 fail인데, integration을 위해 system call을 하지만
실제 SSDManager.exe가 없어서 발생하고 있습니다.
참고 바랍니다 :)

Running main() from C:\Users\User\source\repos\SSDTestShell\packages\gmock.1.11.0\lib\native\src\gtest\src\gtest_main.cc
[==========] Running 14 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 14 tests from TestShellFixture
[ RUN      ] TestShellFixture.execute_invalid_command
[       OK ] TestShellFixture.execute_invalid_command (0 ms)
[ RUN      ] TestShellFixture.execute_write
[       OK ] TestShellFixture.execute_write (0 ms)
[ RUN      ] TestShellFixture.execute_read
[       OK ] TestShellFixture.execute_read (0 ms)
[ RUN      ] TestShellFixture.execute_exit
[       OK ] TestShellFixture.execute_exit (0 ms)
[ RUN      ] TestShellFixture.execute_help
[       OK ] TestShellFixture.execute_help (0 ms)
[ RUN      ] TestShellFixture.write_invalid_argument1
[       OK ] TestShellFixture.write_invalid_argument1 (0 ms)
[ RUN      ] TestShellFixture.write_invalid_argument2
[       OK ] TestShellFixture.write_invalid_argument2 (0 ms)
[ RUN      ] TestShellFixture.read_invalid_argument1
[       OK ] TestShellFixture.read_invalid_argument1 (0 ms)
[ RUN      ] TestShellFixture.read_invalid_argument2
[       OK ] TestShellFixture.read_invalid_argument2 (0 ms)
[ RUN      ] TestShellFixture.fullWrite_fail
C:\Users\User\source\repos\SSDTestShell\TestShellTest\TestShellTest.cpp(97): error: Expected equality of these values:
  mock.fullWrite("")
    Which is: 0
  -1
[  FAILED  ] TestShellFixture.fullWrite_fail (0 ms)
[ RUN      ] TestShellFixture.fullWrite_pass
C:\Users\User\source\repos\SSDTestShell\TestShellTest\TestShellTest.cpp(101): error: Actual function call count doesn't match EXPECT_CALL(mock, write(_))...
         Expected: to be called 100 times
           Actual: never called - unsatisfied and active
[  FAILED  ] TestShellFixture.fullWrite_pass (0 ms)
[ RUN      ] TestShellFixture.fullRead_pass
C:\Users\User\source\repos\SSDTestShell\TestShellTest\TestShellTest.cpp(108): error: Actual function call count doesn't match EXPECT_CALL(mock, read(_))...
         Expected: to be called 100 times
           Actual: never called - unsatisfied and active
[  FAILED  ] TestShellFixture.fullRead_pass (0 ms)
[ RUN      ] TestShellFixture.testApp1_pass
unknown file: error: C++ exception with description "File is not opened" thrown in the test body.
C:\Users\User\source\repos\SSDTestShell\TestShellTest\TestShellTest.cpp(117): error: Actual function call count doesn't match EXPECT_CALL(mock, fullRead())...
         Expected: to be called once
           Actual: never called - unsatisfied and active
[  FAILED  ] TestShellFixture.testApp1_pass (2 ms)
[ RUN      ] TestShellFixture.testApp2_pass
C:\Users\User\source\repos\SSDTestShell\TestShellTest\TestShellTest.cpp(126): error: Actual function call count doesn't match EXPECT_CALL(mock, read(_))...
         Expected: to be called 6 times
           Actual: never called - unsatisfied and active
C:\Users\User\source\repos\SSDTestShell\TestShellTest\TestShellTest.cpp(123): error: Actual function call count doesn't match EXPECT_CALL(mock, write(_))...
         Expected: to be called 36 times
           Actual: never called - unsatisfied and active
[  FAILED  ] TestShellFixture.testApp2_pass (2 ms)
[----------] 14 tests from TestShellFixture (18 ms total)

[----------] Global test environment tear-down
[==========] 14 tests from 1 test suite ran. (19 ms total)
[  PASSED  ] 9 tests.
[  FAILED  ] 5 tests, listed below:
[  FAILED  ] TestShellFixture.fullWrite_fail
[  FAILED  ] TestShellFixture.fullWrite_pass
[  FAILED  ] TestShellFixture.fullRead_pass
[  FAILED  ] TestShellFixture.testApp1_pass
[  FAILED  ] TestShellFixture.testApp2_pass

 5 FAILED TESTS

### CODE REVIEW

File: TestShell/TestShell.cpp [hunk 1]
```diff
@@ -3,8 +3,11 @@
 #include <iostream>
 #include <sstream>
 #include <fstream>
+
 #include "TestShell.h"
 
+#include "../SSDManager/FileManager.cpp"
```
Comment (jiyeon-08) [inline TestShell/TestShell.cpp:9]: 오프라인에서 논의했던 것처럼 이 부분은 추후 TestShell에 FileManager를 따로 넣어서 헤더를 include 하는걸로 변경 부탁드립니다!!

File: TestShell/TestShell.cpp [hunk 2]
```diff
@@ -120,25 +133,103 @@ int TestShell::read(const std::string& arg) {
...
+        if (result != write_value) {
+            test_passed = false;
+            break;
+        }
+    }
     return 0;
 }
 
 int TestShell::testApp2(void) {
+    std::string write_value = "0xAAAABBBB";
```
Comment (Kanaish) [inline TestShell/TestShell.cpp:196]: testApp2를 위한 상수로 선언해서 사용하는 것으로 변경하면 좋을 것 같습니다.

File: TestShell/TestShell.cpp [hunk 3]
```diff
@@ -120,25 +133,103 @@ int TestShell::read(const std::string& arg) {
...
+        this->read(cmd);
+
+        std::cout << file_manager->read("result.txt") << std::endl;
+    }
+
     return 0;
 }
 
 int TestShell::testApp1(void) {
+    std::string write_value = "0xAAAABBBB"; // ������ ��
```
Comment (Kanaish) [inline TestShell/TestShell.cpp:173]: testApp1를 위한 상수로 선언해서 사용하는 것으로 변경하면 좋을 것 같습니다.

File: TestShell/TestShell.cpp [hunk 4]
```diff
@@ -120,25 +133,103 @@ int TestShell::read(const std::string& arg) {
...
+    for (int i = 0; i < 30; ++i) {
+        for (int lba = 0; lba <= 5; ++lba) {
+            std::string arg = std::to_string(lba) + " " + write_value;
+            if (write(arg) != 0) {
+                return -1;
+            }
+        }
+    }
+
+    write_value = "0x12345678";
```
Comment (Kanaish) [inline TestShell/TestShell.cpp:207]: testApp2를 위한 상수로 선언해서 사용하는 것으로 변경하면 좋을 것 같습니다.

File: TestShell/TestShell.cpp [hunk 5]
```diff
@@ -120,25 +133,103 @@ int TestShell::read(const std::string& arg) {
...
+    }
+
     return 0;
 }
 
-int TestShell::fullRead(void) {
+int TestShell::fullRead() {
+    FileManager* file_manager = new FileManager();
+
+    for (int i = 0; i < 100; ++i) {
```
Comment (jiyeon-08) [inline TestShell/TestShell.cpp:161]: 매직넘버 상수로 변경하시면 좋을 것 같습니다~

**Review by jiyeon-08 — APPROVED** [review]
수고하셨습니다~!

File: TestShell/TestShell.cpp (hunk 1 above)
Comment (Kanaish) [reply TestShell/TestShell.cpp:9]: 넵 알겠습니다 :)

**Review by bsjn16sw — CHANGES_REQUESTED** [review]
SSDManager 자체적으로 테스트 완료된 이후, TestShell과도 테스트 진행해보았습니다.

write()와 read()에서 system()을 통해 시스템 콜 날리실 때 SSDManager.exe의 경로 변경 부탁드립니다! (TestShell.exe와 SSDManager.exe가 같은 폴더에 있습니다.)

AS-IS: `std::string cmd = "../x64/Debug/SSDManager.exe w ";`
TO-BE: `std::string cmd = "SSDManager.exe w "; `

AS-IS: `std::string cmd = "../x64/Debug/SSDManager.exe r ";`
TO-BE: `std::string cmd = "SSDManager.exe r "; `



그리고 read() 에서 result.txt 파일을 보여주는 부분도 경로 변경 부탁드립니다.
AS-IS: `cmd = "type result.txt";`
TO-BE: `cmd = "type ../../resources/result.txt";`

File: TestShell/TestShell.cpp (hunk 4 above)
Comment (jiyeon-08) [reply TestShell/TestShell.cpp:207]: 넵 refactoring 시 수정하도록 하겠습니다~

File: TestShell/TestShell.cpp (hunk 3 above)
Comment (jiyeon-08) [reply TestShell/TestShell.cpp:173]: 넵 refactoring 시 수정하도록 하겠습니다~~!

**Comment by Kanaish** [comment]
32c708be4b2a1fb68465459a4fce8d1950069f66
로 SSDManager와 result.txt 파일 path 변경했습니다.

**Comment by Kanaish** [comment]
> [32c708b](https://github.com/Kanaish/SSDTestShell/commit/32c708be4b2a1fb68465459a4fce8d1950069f66)
> 로 SSDManager와 result.txt 파일 path 변경했습니다.

https://github.com/Kanaish/SSDTestShell/commit/32c708be4b2a1fb68465459a4fce8d1950069f66
로 SSDManager와 result.txt 파일 path 변경했습니다.

**Comment by Kanaish** [comment]
> SSDManager 자체적으로 테스트 완료된 이후, TestShell과도 테스트 진행해보았습니다.
> 
> write()와 read()에서 system()을 통해 시스템 콜 날리실 때 SSDManager.exe의 경로 변경 부탁드립니다! (TestShell.exe와 SSDManager.exe가 같은 폴더에 있습니다.)
> 
> AS-IS: `std::string cmd = "../x64/Debug/SSDManager.exe w ";` TO-BE: `std::string cmd = "SSDManager.exe w "; `
> 
> AS-IS: `std::string cmd = "../x64/Debug/SSDManager.exe r ";` TO-BE: `std::string cmd = "SSDManager.exe r "; `
> 
> 그리고 read() 에서 result.txt 파일을 보여주는 부분도 경로 변경 부탁드립니다. AS-IS: `cmd = "type result.txt";` TO-BE: `cmd = "type ../../resources/result.txt";`

https://github.com/Kanaish/SSDTestShell/commit/32c708be4b2a1fb68465459a4fce8d1950069f66
로 SSDManager와 result.txt 파일 path 변경했습니다.

**Comment by Kanaish** [comment]
> > SSDManager 자체적으로 테스트 완료된 이후, TestShell과도 테스트 진행해보았습니다.
> > write()와 read()에서 system()을 통해 시스템 콜 날리실 때 SSDManager.exe의 경로 변경 부탁드립니다! (TestShell.exe와 SSDManager.exe가 같은 폴더에 있습니다.)
> > AS-IS: `std::string cmd = "../x64/Debug/SSDManager.exe w ";` TO-BE: `std::string cmd = "SSDManager.exe w "; `
> > AS-IS: `std::string cmd = "../x64/Debug/SSDManager.exe r ";` TO-BE: `std::string cmd = "SSDManager.exe r "; `
> > 그리고 read() 에서 result.txt 파일을 보여주는 부분도 경로 변경 부탁드립니다. AS-IS: `cmd = "type result.txt";` TO-BE: `cmd = "type ../../resources/result.txt";`
> 
> [32c708b](https://github.com/Kanaish/SSDTestShell/commit/32c708be4b2a1fb68465459a4fce8d1950069f66) 로 SSDManager와 result.txt 파일 path 변경했습니다.

추가로 read시 system call로 파일 read했던 것을 filemanager를 통해 수행하도록 변경했습니다.

**Review by bsjn16sw — APPROVED** [review]
변경해주신 것 확인 완료했습니다! Pull 땡겨서 SSDManager.exe 정상 호출되는지 확인해보겠습니다.!🙂

**Pull request merged**